src/transport/FabricTable.cpp

```diff
@@ -250,10 +250,23 @@ CHIP_ERROR FabricInfo::SetRootCert(const ByteSpan & cert)
     }
     VerifyOrReturnError(mRootCert != nullptr, CHIP_ERROR_NO_MEMORY);
     VerifyOrReturnError(CanCastTo<uint16_t>(cert.size()), CHIP_ERROR_INVALID_ARGUMENT);
-    mRootCertLen          = static_cast<uint16_t>(cert.size());
+    mRootCertLen = static_cast<uint16_t>(cert.size());
+
+    // Find root key ID
+    ChipCertificateData certData;
+    ReturnErrorOnFailure(DecodeChipCert(cert, certData));
+    VerifyOrReturnError(certData.mAuthKeyId.size() <= sizeof(mRootKeyId), CHIP_ERROR_INVALID_ARGUMENT);
+
+    memcpy(mRootKeyId, certData.mAuthKeyId.data(), certData.mAuthKeyId.size());
+    mRootKeyIdLen = certData.mAuthKeyId.size();
+
     mRootCertAllocatedLen = (mRootCertLen > mRootCertAllocatedLen) ? mRootCertLen : mRootCertAllocatedLen;
     memcpy(mRootCert, cert.data(), mRootCertLen);
 
+    Encoding::LittleEndian::BufferWriter bbuf(mRootPubkey, mRootPubkey.Length());
```

Comment:
This is somewhat confusing, in that LittleEndian is namespaced, but the information touched is actually all big-endian, and in any case, just buffer writes, so there is no endianness usage to Put() with a byte buffer.

This usage of BufferWriter could be replaced with a new helper to copy a span into another span with size check. It would be cheaper in runtime and more obvious functionally.